subsys/bluetooth/mesh/vnd/dm_srv.c

```diff
@@ -0,0 +1,474 @@
+	bt_mesh_model_msg_init(&msg, BT_MESH_DM_SRV_OP);
+	net_buf_simple_add_u8(&msg, BT_MESH_DM_SYNC_OP);
+	net_buf_simple_add_u8(&msg, method);
+	net_buf_simple_add_u8(&msg, timeout);
+	(void)bt_mesh_model_send(model, ctx, &msg, NULL, NULL);
+}
+
+static int handle_cfg(struct bt_mesh_model *model, struct bt_mesh_msg_ctx *ctx,
+		      struct net_buf_simple *buf)
+{
+	struct bt_mesh_dm_srv *srv = model->user_data;
+
+	if (!buf->len) {
+		cfg_status_send(model, ctx, 0);
+		return 0;
+	}
+
+	uint8_t ttl = net_buf_simple_pull_u8(buf);
+	uint8_t timeout = net_buf_simple_pull_u8(buf);
+	uint8_t delay = net_buf_simple_pull_u8(buf);
+
+	if ((ttl == 1 || ttl > 127) || timeout < 10) {
+		BT_WARN("Invalid params for configuring DM");
+		cfg_status_send(model, ctx, EBADMSG);
+		return -EBADMSG;
+	}
+
+	srv->cfg.ttl = ttl;
+	srv->cfg.timeout = timeout;
+	srv->cfg.delay = delay;
+
+	store(srv);
+
+	cfg_status_send(model, ctx, 0);
+	return 0;
+}
+
+static int handle_start(struct bt_mesh_model *model, struct bt_mesh_msg_ctx *ctx,
+			struct net_buf_simple *buf)
+{
+	int err = 0;
+	struct bt_mesh_dm_srv *srv = model->user_data;
+	uint8_t timeout = srv->cfg.timeout;
+	uint8_t delay = srv->cfg.delay;
+	uint16_t temp = net_buf_simple_pull_le16(buf);
+	uint8_t tid = net_buf_simple_pull_u8(buf);
+
+	if (tid_check_and_update(&srv->prev_transaction, tid, ctx) != 0) {
+		/* Ignore duplicate message */
+		return 0;
+	}
+
+	if (srv->is_busy) {
+		BT_ERR("Failed to start DM. Measurment already in progress");
+		err = EBUSY;
+		goto rsp;
+	}
+
+	srv->target_addr = temp & 0x7fff;
+
+	enum dm_ranging_mode dm_method = (enum dm_ranging_mode)(temp >> 15);
+	struct bt_mesh_msg_ctx rx_ctx = {
+		.net_idx = ctx->net_idx,
+		.addr = srv->target_addr,
+		.app_idx = ctx->app_idx,
+		.send_ttl = srv->cfg.ttl,
+	};
+
+	/* If optional cfg params is available, overwrite default ones */
+	if (buf->len == 3) {
+		rx_ctx.send_ttl = net_buf_simple_pull_u8(buf);
+		timeout = net_buf_simple_pull_u8(buf);
+		delay = net_buf_simple_pull_u8(buf);
+	}
+
+	if ((rx_ctx.send_ttl == 1 || rx_ctx.send_ttl > 127) || (timeout < 10) || !rx_ctx.addr) {
+		BT_WARN("Invalid params for starting DM");
+		err = EBADMSG;
+		goto rsp;
+	}
+
+	struct dm_request req = {
+		.role = DM_ROLE_REFLECTOR,
+		.ranging_mode = dm_method,
+		.rng_seed = ((bt_mesh_primary_addr() + srv->model->elem_idx) << 16) +
+			srv->target_addr,
+		.start_delay_us = delay
+	};
+
+	BT_INFO("Starting %s DM for target node 0x%04x",
+		dm_method == DM_RANGING_MODE_RTT ? "RTT" : "MCPD", dm_srv->target_addr);
+
+	srv->is_busy = true;
+	k_sem_reset(&dm_srv->dm_ready_sem);
+	err = dm_request_add(&req);
+	if (err) {
+		BT_ERR("DM add request failed: %d", err);
+		srv->is_busy = false;
+		goto rsp;
+	}
+
+	sync_send(model, &rx_ctx, dm_method, timeout);
+	srv->rsp_ctx = *ctx;
+	srv->current_role = DM_ROLE_REFLECTOR;
+	k_work_reschedule(&srv->timeout, K_MSEC(timeout * 100));
+	return 0;
```

Comment:
Can we just not respond to an acked message like this? Or is the idea that the procedure will be fast enough to respond in a timely manner?

Reply:
That is the idea. I've asked @omkar the same thing

Reply:
Yes, the procedure is fast enough. Generally acked message timeout is 1 to 2 seconds. The procedure completes much before that.